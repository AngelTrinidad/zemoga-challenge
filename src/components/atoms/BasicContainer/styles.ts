import {StyleSheet} from 'react-native';

import {DEFAULT_SPACES} from '~constants/styles';

const styles = StyleSheet.create({
  safeArea: {
    flex: 1,
  },
  content: {
    padding: DEFAULT_SPACES.paddingScreen,
  },
});

export default styles;
