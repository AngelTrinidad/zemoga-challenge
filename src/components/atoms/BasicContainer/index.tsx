import React, {FC} from 'react';
import {ViewProps} from 'react-native';

import styles from './styles';

import {ViewFlex} from '~components/atoms';
import {useTheme} from '~contexts';

export interface BasicContainerProps extends ViewProps {
  background?: 'primary' | 'secondary';
}

const BasicContainer: FC<BasicContainerProps> = ({
  children,
  style,
  background = 'primary',
  ...props
}) => {
  const {
    colors: {screen},
  } = useTheme();

  const contentStyle = {
    backgroundColor: screen[background],
  };

  return (
    <ViewFlex style={[styles.content, contentStyle, style]} {...props}>
      {children}
    </ViewFlex>
  );
};

export default BasicContainer;
