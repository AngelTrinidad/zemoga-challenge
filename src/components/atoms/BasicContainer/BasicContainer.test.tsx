import React from 'react';
import {Text} from 'react-native';

import {render} from '@testing-library/react-native';

import BasicContainer from '.';

describe('<BasicContainer />', () => {
  let component: any;

  beforeEach(() => {
    component = render(
      <BasicContainer>
        <Text>Hello</Text>
      </BasicContainer>,
    );
  });

  it('renders correctly', () => {
    expect(component).toBeDefined();
  });

  it('has same snapshot', () => {
    expect(component.toJSON()).toMatchSnapshot();
  });
});
