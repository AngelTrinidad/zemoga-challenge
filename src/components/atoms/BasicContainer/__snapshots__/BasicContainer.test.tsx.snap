// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<BasicContainer /> should has same snapshot 1`] = `
<View
  style={
    [
      {
        "flex": 1,
      },
      undefined,
      [
        {
          "padding": 22.857142857142858,
        },
        {
          "backgroundColor": "#F8F8F8",
        },
        undefined,
      ],
    ]
  }
>
  <Text>
    Hello
  </Text>
</View>
`;
